src/dotnet/src/Mil.Navy.Nrl.Norm/NormSession.cs

```diff
@@ -0,0 +1,918 @@
+﻿using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace Mil.Navy.Nrl.Norm
+{
+    /// <summary>
+    /// NORM transport session
+    /// </summary>
+    public class NormSession
+    {
+        /// <summary>
+        /// The special value NORM_SESSION_INVALID is used to refer to invalid session references.
+        /// </summary>
+        public const int NORM_SESSION_INVALID = 0;
+        /// <summary>
+        /// A dictionary of NORM sessions with their respective handles.
+        /// </summary>
+        private static Dictionary<long, NormSession> _normSessions = new Dictionary<long, NormSession>();
+        /// <summary>
+        /// The NormSessionHandle type is used to reference the NORM transport session.
+        /// </summary>
+        private long _handle;
+
+        /// <summary>
+        /// Used for the application's participation in the NormSession.
+        /// </summary>
+        public long LocalNodeId
+        {
+            get => NormGetLocalNodeId(_handle);
+        }
+
+        /// <summary>
+        /// The report interval.
+        /// </summary>
+        public double ReportInterval 
+        {
+            get => NormGetReportInterval(_handle); 
+            set => NormSetReportInterval(_handle, value); 
+        }
+
+        /// <summary>
+        /// Transmission rate (in bits per second (bps)) limit used for NormSender transmissions.
+        /// </summary>
+        public double TxRate
+        {
+            get => NormGetTxRate(_handle);
+            set => NormSetTxRate(_handle, value);
+        }
+
+        /// <summary>
+        /// Group round-trip timing.
+        /// </summary>
+        public double GrttEstimate
+        {
+            get => NormGetGrttEstimate(_handle);
+            set => NormSetGrttEstimate(_handle, value);
+        }
+
+        /// <summary>
+        /// Internal constructor of NormSession.
+        /// </summary>
+        /// <param name="handle">Used to identify application in the NormSession.</param>
+        /// <remarks>The handle and NormSession are added to the dictionary of NORM sessions.</remarks>
+        internal NormSession(long handle)
+        {
+            _handle = handle;
+            lock (_normSessions)
+            {
+                _normSessions.Add(handle, this);
+            }     
+        }
+
+        /// <summary>
+        /// Get a specified NormSession from the dictionary of NORM sessions.
+        /// </summary>
+        /// <param name="handle">Specifies the session to return.</param>
+        /// <returns>Returns a NormSession.</returns>
+        [MethodImpl(MethodImplOptions.Synchronized)]
```

Comment:
No need for this. Just `lock` on `_normSessions`.

Reply:
Changed `GetSession` to use `TryGetValue`